src/end.sol

```diff
@@ -283,11 +312,28 @@ contract End is LibNote {
         tag[ilk] = wdiv(spot.par(), uint(pip.read()));
     }
 
+    function halt(bytes32 ilk, uint256 id) external note {
+        require(tag[ilk] != 0, "End/tag-ilk-not-defined");
+
+        (address _clip,,,) = dog.ilks(ilk);
+        ClipLike clip = ClipLike(_clip);
+        (, uint rate,,,) = vat.ilks(ilk);
+        (, uint256 tab, uint256 lot, address usr,,) = clip.sales(id);
+        
+        vat.suck(address(vow), address(vow),  tab);
```

Comment:
@kmbarry1 @lucas-manuel I know this same line is also in `skip` so there has to be some reason to be there but I can't get why (haven't done a deep analysis though).
Probably there is a an obvious reason I'm missing, but executing `vow.heal(tab)` after `skip` or `halt` reverts this action so no getting exactly what is the purpose. There is probably a scenario that I'm not considering where this is necessary.

Reply:
The `grab` call at the end is going to subtract `tab` from the `Vow's` `sin` balance. However, it's not guaranteed that the `Vow` will actually have that much `sin`, meaning you might get an underflow that reverts the transaction. This `suck` call ensures that the necessary `sin` is present in the `Vow's` balance, with equivalent `dai` so that the net effect on `Vat.dai(vow) - Vat.sin(vow)` is the same. And the reason you want to add to the system balance in the first place is to cancel out the `sin` that was assigned to the `Vow` when the position was liquidated. It's not an exact cancellation because it now includes the liquidation penalty; this is also by design, since the restored Vault also has the liquidation penalty added to its debt, meaning that extra collateral will be withheld by `skim` to cover that penalty, which now goes to DAI holders because it is in the surplus buffer.

Reply:
Yes perfect, got it, that possible underflow is what I was missing. Thanks!